.github/workflows/test_workflow_scripts/golang/gin_mongo/golang-docker-macos.sh

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
+
+# macOS variant for gin-mongo (docker run)
+set -euo pipefail
+
+# Isolate keploy home per run to avoid cross-job collisions on a single self-hosted runner
+export KEPLOY_HOME_ROOT="${TMPDIR:-/tmp}/keploy-run-${GITHUB_RUN_ID:-$$}-${GITHUB_JOB:-gin-mongo}-$(date +%s)"
+export HOME="$KEPLOY_HOME_ROOT/home"
+mkdir -p "$HOME"
+
+source ./../../.github/workflows/test_workflow_scripts/test-iid.sh
+
+# cleanup() {
+#     set +e
+#     # Stop and remove docker resources
+#     docker rm -f ginApp || true
+#     docker rm -f ginApp_test || true
+#     docker stop mongoDb || true
+#     docker rm mongoDb || true
+#     docker network rm keploy-network || true
+#     # Remove generated artifacts
+#     rm -rf keploy/ || true
+#     rm -f echoApp.txt ginApp_*.txt ginApp_test.txt || true
+#     # Remove keploy home override
+#     rm -rf "$KEPLOY_HOME_ROOT" || true
+# }
+# trap cleanup EXIT INT TERM
+
+# Start mongo before starting keploy.
+docker network create keploy-network || true
+docker run --name mongoDb --rm --net keploy-network -p 27017:27017 -d mongo
+
+# Generate the keploy-config file.
+sudo -E env HOME="$HOME" PATH=$PATH "$RECORD_BIN" config --generate
+
+# Update the global noise to ts.
+config_file="./keploy.yml"
+sed -i '' 's/global: {}/global: {"body": {"ts":[]}}/' "$config_file"
+
+# Remove any preexisting keploy tests and mocks.
+sudo rm -rf keploy/
+docker logs mongoDb &
+
+# Start keploy in record mode.
+docker build -t gin-mongo .
+docker rm -f ginApp 2>/dev/null || true
+
+container_kill() {
+    pid=$(pgrep -n keploy || true)
+    if [ -n "$pid" ]; then
+      echo "$pid Keploy PID"
+      echo "Killing keploy"
+      sudo kill $pid || true
+    fi
+}
+
+send_request(){
+    sleep 10
+    app_started=false
+    while [ "$app_started" = false ]; do
+        if curl -X GET http://localhost:8080/CJBKJd92; then
+            app_started=true
+        fi
+        sleep 3
+    done
+    echo "App started"
+    # Start making curl calls to record the testcases and mocks.
+    curl --request POST \
+      --url http://localhost:8080/url \
+      --header 'content-type: application/json' \
+      --data '{
+      "url": "https://google.com"
+    }'
+
+    curl --request POST \
+      --url http://localhost:8080/url \
+      --header 'content-type: application/json' \
+      --data '{
+      "url": "https://facebook.com"
+    }'
+
+    curl -X GET http://localhost:8080/CJBKJd92
+
+    # Wait for 5 seconds for keploy to record the tcs and mocks.
+    sleep 5
+    container_kill
+    wait || true
+}
+
+for i in {1..2}; do
+    container_name="ginApp_${i}"
+    send_request &
+    sudo -E env HOME="$HOME" PATH=$PATH "$RECORD_BIN" record -c "docker run -p8080:8080 --net keploy-network --rm --name $container_name gin-mongo" --container-name "$container_name"    &> "${container_name}.txt" || true
```

Comment:
We are thinking to print the logs irrespective of the success and failure ?

Reply:
deleted this script